[AST] Prefer the 'macOS' spelling over 'OSX' when printing the platform kind.

nathawes opened the pull request:
This affects module interfaces, interface generation in sourcekitd, and diagnostics. Also fixes a fixit that was assuming the 'OSX' spelling when computing the source range to replace, and results in PlatformKind::OSX and PlatformKind::OSXApplicationExtension being renamed to PlatformKind::macOS and PlatformKind::macOSApplicationExtension too.

Resolves rdar://problem/64667960

Commit:
[AST] Prefer the 'macOS' spelling over 'OSX' when printing the platform kind.

This affects module interfaces, interface generation in sourcekitd, and
diagnostics. Also fixes a fixit that was assuming the 'OSX' spelling when
computing the source range to replace.

Resolves rdar://problem/64667960

nathawes commented:
@swift-ci please test

nathawes commented:
@swift-ci please test

nathawes commented:
@swift-ci please test

nathawes commented:
@swift-ci please test OS X Platform

nathawes commented:
@swift-ci please test

nathawes commented:
@swift-ci please test

nathawes commented:
@swift-ci please test

brentdax commented on lib/AST/PlatformKind.cpp:
```diff
@@ -24,6 +24,15 @@
 using namespace swift;
 
 StringRef swift::platformString(PlatformKind platform) {
+  // FIXME: Update PlatformKinds.def to use the macOS spelling by default.
```
TBH, it's probably a good idea to just bite the bullet and do this, rather than continuing to add more special cases everywhere.

nathawes replied:
Done. @brentdax do you think this change would be ok for 5.3, or too risky?

brentdax replied:
I don’t think there’s much additional risk from implementing it this way:

* By far the riskiest thing in this PR is changing the string we print into machine-readable files, because we may not have test coverage proving that the compiler accepts “macOS” everywhere it would accept “OSX”, and we certainly can’t prove that all non-compiler tools that work with these files will accept it. But both implementations do this, so changing PlatformKinds.def has no bearing on it.

* Changing the contents of PlatformKinds.def creates additional risk in the places where we actually include PlatformKinds.def, but it’s trivial to audit these (there are only four) and see that they’re all correct.

    * We should verify that there’s test coverage of both `”OSX”` and `”macOS”` being fed into `swift::platformFromString()`, but I have little doubt that there is.

* There’s no additional risk at places where we merely use `PlatformKind::OSX` because the compiler verifies that all of those are changed to `PlatformKind::macOS`. (However, this could cause mis-merges/merge conflicts for a little while.)

nathawes replied:
Thanks Brent! Your first point does sound a little worrying. I think I'll see if I can make the spelling change specific to sourcekitd's interface generation for 5.3 to avoid that risk and leave this more comprehensive spelling change as master-only.

Commit:
[AST] Rename PlatformKind::OSX to PlatformKind::macOS

Because the names are coming from a .def file used for printing too, this
simplifies the printing logic as well.

nathawes commented:
@swift-ci please test

beccadax reviewed (approved):
Wonderful!

nathawes commented:
@swift-ci please smoke test
